### PR TITLE
Export category

### PR DESCRIPTION
**Pull request by filafb**
#### What problem is this solving?

This PR allows user to download translations for Categories (name, title and description). It gives user the ability to download all categories or only the actives ones.

#### The main files to checkout:

_This PR adds the node and graphql builder._

__Backend__ 

The main entry files can be found on `./graphql/*` and `./node/index.ts`.

_./node/clients/catalog.ts_

This file leverages `vtex.catalog-graphql@1.x` to get all category ids and category translations.

The method `getCategoriesId` fetches all ids. To perform that, since we don't have a way to get all IDs at once, it fetches the first page of categories. Based on the pagination info, it loops over the pages and fetch all other ids, page by page. Not very performative though. To make it a little better, all the calls are made async, and resolved inside a `Promise.all`. 

The other public method `getTranslation` fetches a single translation based on the passed id. It forwards the headers passed from the front end call.

_./node/resolvers/category.ts_

The resolver for `categoryTranslations` query gets all the category ids, then get all their translation, one by one. Again, not very performative. It returns the array of promises, so the responses are handled by the `Category` resolver.

The `Category` resolver receives the data as it come from the HTTP call and get its value from the `root` argument. This way, we avoid one more loop over all the entries inside the `categoryTranslations` to extract the category object.

__Frontend__

_./react/components/CategoryTranslation/CategoryTranslation.tsx_

The new [useLazyQuery](https://gitlab.com/vtex_europe/catalog-translation/-/merge_requests/new/diffs?merge_request%5Bsource_branch%5D=export-category&merge_request%5Bsource_project_id%5D=23444664&merge_request%5Btarget_branch%5D=master&merge_request%5Btarget_project_id%5D=23444664#0aeddcc735d2cca2b33fc4493f6d4ef15b54890a_23_44) touches the new graphql functions in the backend. It passes the locale we want the translation for the headers request.

The useEffect listening to data from the above query calls a new helper function to parse json to xls.

_react/utils/index.ts_

The new `parseJSONToXLS` function uses [xlsx](https://yarnpkg.com/package/xlsx) to convert json to xls and prompt the download for the user. It seems a very stable lib (4M downloads in the last 30 days) and the US team is using it in the [quick order app](https://github.com/vtex-apps/quickorder/blob/4b04f290c66fe6782573cd54d40f80d5f3095b5f/react/UploadBlock.tsx#L17).

#### How to test it?

All workspaces below have the code linked:

[Muji](https://exportcategory--muji.myvtex.com/admin/catalog-translation/category)

[Ametller](https://exportcategory--powerplanet.myvtex.com/admin/catalog-translation/category)

[Powerplanet](https://exportcategory--powerplanet.myvtex.com/admin/catalog-translation/category)

#### Screenshots or example usage:

recording (1)

### CODE REVIEW

File: react/components/CategoryTranslation/CategoryTranslation.tsx [hunk 1]
```diff
@@ -32,39 +63,123 @@ const CategoryTranslation: FC = () => {
...
+    })
+  }
+
+  useEffect(() => {
+    // eslint-disable-next-line vtex/prefer-early-return
+    if (data) {
+      parseJSONToXLS(data.categoryTranslations, {
+        fileName: 'category-data',
+        sheetName: 'category_data',
+      })
```
Comment (cesarocampov) [inline react/components/CategoryTranslation/CategoryTranslation.tsx:80]: @filafb Is it possible to have the fileName be a construct of 'category-data'+locale?
so if I click on it-IT my file will be named category-data-it-IT etc

**Review by cesarocampov — APPROVED** [review]

**Comment by filafb** [comment]
The update version is linked [here](https://exportcategory--powerplanet.myvtex.com/admin/catalog-translation/category).

Changes:
1. Include the locale in the file name;
2. Change the extension to xlsx;
3. Allow user to download the same file in a sequence.

**Comment by cesarocampov** [comment]
@filafb amazing!

**Comment by filafb** [comment]
@cesarocampov I also changed the flag to avoid releasing a new version. This way we can release category and product under the same minor.

**Review by edyespinal — APPROVED** [review]

**Comment by edyespinal** [comment]
Nice work @filafb!

**Comment by filafb** [comment]
> Nice work @filafb!
> 
> Only thing I think could be improved is more related to UX.
> Should we find a way to let the users know that they can export one or all categories?
> I was using the Search input and didn't realised that I could click the export and export every category.

Agree. Cesar also asked a different UX in another branch I am working on. Will try to have it done over there.

**Pull request merged**